packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -136,11 +136,25 @@ export function recmaJsxRewrite(options = {}) {
             const fullId = ids.join('.')
             const id = name.name
 
+            const isInScope = inScope(currentScope, id)
+
             if (!own.call(fnScope.references, fullId)) {
-              fnScope.references[fullId] = {node, component: true}
+              const parentScope = /** @type {Scope|null} */ (
+                currentScope.parent
+              )
+              const parentNode = parentScope && parentScope.node
```

Comment:
```suggestion

```

I think the `node` is always there, assuming `parentScope` is non-nully, right? If so, we can move this check to the condition below?